server/bin/test/test_pbench_server_put_shim_001.py

```diff
@@ -87,53 +83,45 @@ def mock_copy_result_tb(self, token):
         )
         monkeypatch.setattr(CopyResultTb, "copy_result_tb", mock_copy_result_tb)
         res = process_tb(self.config, self.logger, receive_dir)
-        log = str(self.log_output())
+        log = self.get_last_log_line()
 
         assert res == expected_result
         assert log.endswith(expected_error_msg)
 
-    def test_bad_tb(self):
-        """checks Results(namedtuple) return from process_tb module when any
-            kind of error is raised."""
-
-        receive_dir = "server/bin/test/fixtures/upload/bad_tarball"
-        expected_error_msg = Results(nstatus="", ntotal=1, ntbs=0)
-        res = process_tb(self.config, self.logger, receive_dir)
-
-        assert res == expected_error_msg
-
     def test_process_tb(self, monkeypatch):
-        """verify processing of tbs without any failure"""
+        """verify processing of tar balls without any failure"""
 
         def mock_copy_result_tb(self, token):
             return
 
         receive_dir = "server/bin/test/fixtures/upload/tarball"
-        expected_error_msg = Results(
+        expected_result = Results(
             nstatus=f": processed {receive_dir}/tar/log.tar.xz\n", ntotal=1, ntbs=1
         )
         monkeypatch.setattr(CopyResultTb, "copy_result_tb", mock_copy_result_tb)
         res = process_tb(self.config, self.logger, receive_dir)
 
-        assert res == expected_error_msg
+        assert res == expected_result
 
     def test_process_tb_zero(self):
         """verify processing if there are no TBs without any failure"""
 
         receive_dir = "server/bin/test/fixtures/upload/tarball"
-        expected_error_msg = Results(nstatus="", ntotal=0, ntbs=0)
+        expected_result = Results(nstatus="", ntotal=0, ntbs=0)
         res = process_tb(self.config, self.logger, receive_dir)
 
-        assert res == expected_error_msg
+        assert res == expected_result
 
     def test_multiple_process_tb(self, monkeypatch):
-        """verify tbs processing at the time of Failure as well as success"""
+        """verify tar balls processing at the time of Failure as well as success"""
 
         def mock_copy_result_tb(self, token):
             if self.tarball.exists():
                 return
             else:
-                raise RuntimeError(f"Cannot connect to '{self.upload_url}'")
+                raise FileNotFoundError(
+                    f"No such file or directory: '{receive_dir}/bad/bad_tb.tar.xz'"
+                )
```

Comment:
I don't think that raising `FileNotFoundError` is the right thing to do, here (and, I think there are a couple of things wrong with raising it with that specific message!).

First, as the CUT stands at the moment, I believe that `copy_result_tb()` would never be called with a non-existent or inaccessible file, so we should never see this `raise` executed.  That suggests to me that, instead of doing a `raise`, we should be doing something like `assert False, "Unexpected file path"` or `assert False, "Unexpected behavior"`.

Second, the not-found file name shouldn't be hard coded -- it should come from `self.tarball`!  😀  (BTW, the value there now _still_ doesn't match the value that the test checks at line 131.... 🙂)

Finally, since this is a _unit test_, we shouldn't be depending on `md5sum()` going out to the file system to determine that the file does not exist:  we should be mocking that function and having it raise the exception based on the input path that it receives -- we (should) know what paths the CUT is going to pass into the (mocked) function, and so we can make it behave appropriately for what the test requires.

Reply:
Sure. I believe we do not have to raise the FileNotFound Error it will be logged and I can just capture the log and compare with it.